File: Jogando Com a Matemática/Assets/Scripts/LoginAluno.cs
using UnityEngine;
using UnityEngine.UI;
using UnityEngine.Networking;
using System.Collections;

[System.Serializable]
public class LoginResponse
{
    public int id_Aluno;
    public int id_turma;
    public string status;
    public string erro;
}

public class LoginAluno : MonoBehaviour // Verifica o login do aluno atraves de PHP.
{
    public InputField inputMatricula; // Campo de entrada para matrícula
    public InputField inputSenha;    // Campo de entrada para senha
    public Text mensagemErro;        // Texto para exibir mensagens de erro

    public void Login()
    {
        EconomyManager.Instance.currentGold = 0;
        StartCoroutine(LoginCoroutine());
    }

    private IEnumerator LoginCoroutine()
    {
        string url = "http://localhost/projeto/login_aluno.php";

        // Criando o formulário com as credenciais
        WWWForm form = new WWWForm();
        form.AddField("matricula", inputMatricula.text);
        form.AddField("senha", inputSenha.text);

        using (UnityWebRequest request = UnityWebRequest.Post(url, form))
        {
            yield return request.SendWebRequest();

            if (request.result == UnityWebRequest.Result.Success)
            {
                string json = request.downloadHandler.text;
                Debug.Log("Resposta do servidor: " + json);

                try
                {
                    // Deserializando o JSON para um objeto LoginResponse
                    LoginResponse response = JsonUtility.FromJson<LoginResponse>(json);

                    if (response.status == "success")
                    {
                        Debug.Log($"Login bem-sucedido! ID Aluno: {response.id_Aluno}, ID Turma: {response.id_turma}");

                        // Salvando informações no PlayerPrefs
                        PlayerPrefs.SetInt("id_turma", response.id_turma);
                        PlayerPrefs.SetInt("id_Aluno", response.id_Aluno);

                        // Indo para a próxima cena
                        UnityEngine.SceneManagement.SceneManager.LoadScene("Operacao");
                    }
                    else
                    {
                        mensagemErro.text = response.erro;
                        Debug.Log("Login inválido: " + response.erro);
                    }
                }
                catch (System.Exception e)
                {
                    Debug.LogError($"Erro ao processar JSON: {e.Message}");
                    mensagemErro.text = "Erro ao processar a resposta. Tente novamente.";
                }
            }
            else
            {
                Debug.LogError("Erro na requisição: " + request.error);
                mensagemErro.text = "Erro ao conectar ao servidor. Verifique sua conexão.";
            }
        }
    }
}

File: Arquivos Game/Arquivos PHP - (Unity) - htdocs/projeto/login_aluno.php
<?php
$conn = new mysqli('localhost', 'root', '', 'alunos_db');

if ($conn->connect_error) {
    die("Erro de conexão: " . $conn->connect_error);
}

$matricula = isset($_POST['matricula']) ? $_POST['matricula'] : '';
$senha = isset($_POST['senha']) ? $_POST['senha'] : '';

$sql = "SELECT id_Aluno, id_turma, genero FROM aluno WHERE matricula = ? AND senha = ?";
$stmt = $conn->prepare($sql);
$stmt->bind_param("ss", $matricula, $senha);
$stmt->execute();
$result = $stmt->get_result();

$response = new stdClass();

if ($row = $result->fetch_assoc()) {
    $response->id_Aluno = $row['id_Aluno'];
    $response->id_turma = $row['id_turma'];
    $response->genero = $row['genero'];
    $response->status = "success";
} else {
    $response->erro = "Login inválido";
    $response->status = "fail";
}

header('Content-Type: application/json');
echo json_encode($response);

$conn->close();
?>



